sync/firewall/firewall.go

```diff
@@ -83,6 +83,12 @@ func (f *Firewall) IsBannedAddress(remoteAddr string) bool {
 	return f.ipBlocker.IsBanned(ip)
 }
 
+func (f *Firewall) IsDuplicatePeer(peerID peer.ID, remoteAddr string) bool {
```

Comment:
It can merge into `IsBannedAddress` (we can rename it to `IsAddressBanned` for better semantic name)